### PR TITLE
Python API docs - add duckdb.sql, read_csv methods, and split the Python API docs into multiple (hopefully more clear) pages

### PR DESCRIPTION
**Pull request by Mytherin**
This PR improves the Python docs in various aspects:

* Use `duckdb.sql` in examples, instead of creating a connection first
* Greatly simplify the "DuckDB and Polars" guide to make use of the new Polars <> DuckDB integration
* Split the Python API overview into separate pages (overview, data ingestion, result conversion and DB API)
* Add (limited) documentation for the new ingestion functions (`.read_csv`, `.read_json` , `.read_parquet`)

Please take a look, there's a bunch of new text + restructuring here.

### CODE REVIEW

File: docs/api/python/data_ingestion.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+duckdb.read_json('folder/*.json')
+# directly read a JSON file from within SQL
+duckdb.sql("SELECT * FROM 'example.json'")
+# call read_json from within SQL
+duckdb.sql("SELECT * FROM read_json_auto('example.json')")
+```
+
+#### DataFrames & Arrow Tables
+
+DuckDB is automatically able to query a Pandas DataFrame or Arrow object that is stored in a Python variable by name. DuckDB supports querying multiple types of Apache Arrow objects including [tables](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html), [datasets](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Dataset.html), [recordbatchreaders](https://arrow.apache.org/docs/python/generated/pyarrow.ipc.RecordBatchStreamReader.html), and [scanners](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Scanner.html). See the Python [guides](../../guides/index#python-client) for more examples.
```
Comment (Mause) [inline docs/api/python/data_ingestion.md:64]:
```suggestion
DuckDB is automatically able to query a Pandas DataFrame or Arrow object that is stored in a Python variable by name. DuckDB supports querying multiple types of Apache Arrow objects including [tables](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html), [datasets](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Dataset.html), [RecordBatchReaders](https://arrow.apache.org/docs/python/generated/pyarrow.ipc.RecordBatchStreamReader.html), and [scanners](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Scanner.html). See the Python [guides](../../guides/index#python-client) for more examples.
```

File: docs/api/python/data_ingestion.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+# [(1, 'one'), (2, 'two'), (3, 'three'), (4, 'four')]
+```
+
+DuckDB also supports "registering" a DataFrame or Arrow object as a virtual table, comparable to a SQL `VIEW`. This is useful when querying a DataFrame/Arrow object that is stored in another way (as a class variable, or a value in a dictionary). Below is a Pandas example:
+
+If your Pandas DataFrame is stored in another location, here is an example of manually registering it:
+```python
+import duckdb
+import pandas as pd
+my_dictionary = {}
```
Comment (Mause) [inline docs/api/python/data_ingestion.md:80]: What's this dictionary for?

File: docs/api/python/dbapi.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+# retrieve the items again
+con.execute("SELECT * FROM items")
+print(con.fetchall())
+# [('jeans', 20.0, 1), ('hammer', 42.2, 2)]
+```
+
+The `description` property of the connection object contains the column names as per the standard.
+
+DuckDB also supports prepared statements in the API with the `execute` and `executemany` methods. The values may be passed as an additional parameter after a query that contains `?` or `$1` (dollar symbol and a number) placeholders. Using the `?` notation adds the values in the same sequence as passed within the Python parameter. Using the `$` notation allows for values to be reused within the SQL statement based on the number and index of the value found within the Python parameter.
```
Comment (Mause) [inline docs/api/python/dbapi.md:42]: Worth mentioning named parameter support here?

**Review by Mause — CHANGES_REQUESTED** [review]

File: docs/api/python/data_ingestion.md (hunk 2 above)
Comment (Mytherin) [reply docs/api/python/data_ingestion.md:80]: This is just moved from the previous docs - IIRC the dictionary is there to show how we can scan a Pandas DataFrame that is not stored in the global namespace (i.e. inside a dictionary or other container)

File: docs/api/python/dbapi.md (hunk 3 above)
Comment (Mytherin) [reply docs/api/python/dbapi.md:42]: Definitely, perhaps @Tishj can pick that up as part of a different PR

**Comment by Alex-Monahan** [comment]
Thanks for this Mark, it's great! I really like how you have split it apart. I made a few small tweaks.

Is it ok that I made them directly on your branch? I can do a separate branch or use the request changes feature it the future if that is easier!

**Review by Alex-Monahan — APPROVED** [review]

**Comment by Mytherin** [comment]
I don't mind - thanks for having a look/making changes :)

**Pull request merged**